kv/index_iter.go

```diff
@@ -138,17 +138,17 @@ func (c *kvIndex) GenIndexKey(indexedValues []interface{}, h int64) (key []byte,
 		}
 	}
 
-	var encVal []byte
+	var b bytes.Buffer
+	b.WriteString(c.prefix)
```

Comment:
bytes.Buffer maintains a 64 byte pre-alloced slice. So when writing to Buffer, declare a `bytes.Buffer` then write to it will reduce memory alloc in most common cases.
I think the best practice is always use `var b bytes.Buffer` to write, and use `NewBufferString` to read as much as possible.